imports/server/gdriveDocumentWatcher.ts

```diff
@@ -0,0 +1,284 @@
+        });
+      }
+
+      // Clean up the old watch if there was one
+      if (watch) {
+        await DriveClient.gdrive.channels.stop({
+          requestBody: {
+            id: watch.watchId,
+            resourceId: watch.watchResourceId,
+          },
+        });
+      }
+    });
+  }
+}
+
+const NGROK_API_URL = 'http://127.0.0.1:4040';
+
+let warnedAboutWebhooks = false;
+async function discoverWebhookRoot() {
+  const url = Meteor.absoluteUrl();
+  if (!Meteor.isDevelopment) {
+    return url;
+  }
+
+  const parsed = new URL(url);
+  if (parsed.hostname !== 'localhost') {
+    return url;
+  }
+
+  const ngrok = await import('ngrok');
+  const api = new ngrok.NgrokClient(NGROK_API_URL);
+
+  let tunnels;
+  try {
+    tunnels = await api.listTunnels();
+  } catch (e) {
+    if (!warnedAboutWebhooks) {
+      /* eslint-disable no-console */
+      console.warn(e);
+      console.warn();
+      console.warn(
+        'Unable to discover a URL for receiving webhooks. If you want to test incoming Google\n' +
+        'Drive webhooks, then jolly-roger needs to be accessible from the internet. The easiest\n' +
+        'choice is to start ngrok by running "meteor npx ngrok start --none" (we will\n' +
+        'automatically configure the rest).\n' +
+        '\n' +
+        'But if you would prefer to set things up manually, set the ROOT_URL environment\n' +
+        'variable to a publicly accessible URL that routes to this jolly-roger server.'
+      );
+      /* eslint-enable no-console */
+      warnedAboutWebhooks = true;
+    }
+    return undefined;
+  }
+
+  const tunnel = tunnels.tunnels.find((t) => {
+    const parsedTunnel = new URL(t.config.addr);
+    return t.proto === 'https' && parsedTunnel.port === parsed.port;
+  });
+
+  if (tunnel) {
+    return tunnel.public_url;
+  }
+
+  // No tunnel, so we need to try and create one
+  const newTunnel = await api.startTunnel({ name: 'jolly-roger', proto: 'http', addr: url });
+  return newTunnel.public_url;
+}
+
+Meteor.startup(() => {
+  if (Meteor.isTest || Meteor.isAppTest) {
+    return;
+  }
+
+  let watcher: GDriveDocumentWatcher | undefined;
+  let enabledByFeatureFlag = true;
+  const updateWatcher = async () => {
+    const newEnabledByFeatureFlag = !Flags.active('disable.gdrive_watchers');
+    const newRootUrl = await discoverWebhookRoot();
+    const shouldRestart =
+      newRootUrl !== watcher?.rootUrl ||
+      newEnabledByFeatureFlag !== enabledByFeatureFlag;
+    enabledByFeatureFlag = newEnabledByFeatureFlag;
+
+    if (shouldRestart) {
+      watcher?.stop();
+    }
+
+    if (shouldRestart && newRootUrl && enabledByFeatureFlag) {
+      Ansible.log('Starting Google Drive webhooks with public URL', { url: newRootUrl });
+      watcher = new GDriveDocumentWatcher(newRootUrl);
+    }
+  };
+  Meteor.setInterval(updateWatcher, 5 * 1000);
+  FeatureFlags.find({ name: 'disable.gdrive_watchers' }).observe({
+    added: () => { void updateWatcher; },
```

Comment:
Does this actually call `updateWatcher()`?  It looks to me like it's not invoking `updateWatcher` but maybe I'm confused by how promises are awaited

Reply:
Oh no, you're right. I originally wrote it as `added: updateWatcher`, which would have worked, but needed the extra arrow function for the void cast. Pushed a fix.